index.html

```diff
@@ -2200,57 +2199,161 @@ <h2>Sessions</h2>
+
+<p>An <a>endpoint node</a> has an associated <dfn>has proxy
+ configuration</dfn> flag that indicates whether the proxy is already
+ configured. The default value of the flag is true if the endpoint
+ doesn't support proxy configuration, or false otherwise.</p>
+
+<div>
+
+<p>To <dfn>create a session</dfn>, given a JSON
+Object <var>capabilites</var>, and a set <var>flags</var>:
+<ol>
+ <li><p>Let <var>session id</var> be the result of <a>generating a UUID</a>.
+
+ <li><p>Let <var>session</var> be a new <a>session</a> with <a>session
+  ID</a> <var>session id</var>, and <a>HTTP flag</a> <var>flags</var> contains
+  "<code>http</code>".
+
+ <li><p>Let <var>proxy</var> be the result of getting property
+     "<code>proxy</code>" from <var>capabilities</var> and run the
+     substeps of the first matching statement:
+
+       <dl class=switch>
+         <dt><var>proxy</var> is a <a>proxy configuration</a> object</dt>
+         <dd><p>Take implementation-defined steps to set the user agent proxy
+             using the extracted <var>proxy</var> configuration. If the
+             defined proxy cannot be configured return <a>error</a> with
+             <a>error code</a> <a>session not created</a>. Otherwise
+             set the <a>has proxy configuration</a> flag to true.</dd>
+
+         <dt>Otherwise</dt>
+         <dd><a>Set a property</a> of <var>capabilities</var> with name
+           "<code>proxy</code>" and a value that is a new JSON <a>Object</a>.</dd>
+       </dl>
+   </li>
+
+  <li><p>If <var>capabilites</var> has a property named
+  "<code>acceptInsecureCerts</code>", set the <a>endpoint
+  node</a>'s <a>accept insecure TLS</a> flag to the result
+  of <a>getting a property</a> named
+  "<code>acceptInsecureCerts</code>" from <var>capabilities</var>.
+
+  <li><p>If <var>flags</var> <var>contains</var> "<code>http</code>":
+      <ol>
+        <li><p>Let <var>strategy</var> be the result of getting
+            property "<code>pageLoadStrategy</code>"
+            from <var>capabilities</var>. <p>If <var>strategy</var> is
+            a string, set the <a>session</a>’s
+            <a>page loading strategy</a> to <var>strategy</var>.  Otherwise,
+            set the <a>page loading strategy</a> to <i>normal</i> and <a>set a
+              property</a> of <var>capabilities</var> with name
+            "<code>pageLoadStrategy</code>" and value "<code>normal</code>".</p>
+
+        <li><p>Let <var>strictFileInteractability</var> be the result of
+            getting property "<code>strictFileInteractability</code>"
+            from <var>capabilities</var>. If <var>strictFileInteractability</var>
+            is a boolean, set <a>session</a>’s <a>strict file interactability</a> to
+            <var>strictFileInteractability</var>. Otherwise set
+            <a>session</a>’s <a>strict file interactability</a> to false.</p></li>
+
+        <li><p>If <var>capabilities</var> has a property with the key "<code>timeouts</code>":
```

Comment:
So far we don't have any use case for timeouts in BiDi (the client can always implement timeouts when the protocol is non-blocking, as long as no cleanup is required), and I'm keen to avoid timeouts as global state in BiDi.